mono/metadata/icall.c

```diff
@@ -1162,40 +1162,75 @@ ves_icall_System_ValueType_Equals (MonoObject *this_obj, MonoObject *that, MonoA
 			continue;
 		if (mono_field_is_deleted (field))
 			continue;
+		guint8 *this_field = (guint8 *) this_obj + field->offset;
+		guint8 *that_field = (guint8 *) that + field->offset;
+
+#define UNALIGNED_COMPARE(type) \
+			do { \
+				type left, right; \
+				memcpy (&left, this_field, sizeof (type)); \
+				memcpy (&right, that_field, sizeof (type)); \
+				if (left != right) \
+					return FALSE; \
+			} while (0);
```

Comment:
Trailing semicolon probably not desired here.

Reply:
ah, that explains why I _had_ to add the curly braces around the if statement... 😅 

see https://github.com/mono/mono/pull/6403/

Reply:
I like to use "self" instead of "this_obj", assuming "this" is avoided for C++ compat, which I agree is a good idea.